File: 74-search-a-2d-matrix/74-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        for(int i=0; i<matrix.length; i++){
            if(target >= matrix[0][0] && target <= matrix[i][matrix[i].length - 1]){
                int p = 0;
                int q = matrix[i].length -1 ;
                while(p <= q){
                    int t = (p+q)/2;
                    if(target == matrix[i][t]){
                        return true;
                    }
                    else if(target < matrix[i][t]){
                        q = t - 1;
                    }
                    else if(target > matrix[i][t]){
                        p = t + 1;
                    }
                }
            }
        }
        return false;
    }
}

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        for(int i=0;i <matrix.size(); i++){
            if(matrix[i][0] <= target && matrix[i][matrix[i].size() - 1] >= target){
                int low = 0;
                int high = matrix[i].size() - 1;
                while(low <= high){
                    int mid = (low + high)/2;
                    if(matrix[i][mid] == target){
                        return true;
                    }
                    else if(target > matrix[i][mid]){
                        low = mid + 1;
                    }
                    else{
                        high = mid - 1;
                    }
                }
            }
        }
        return false;
    }
};